src/plugins/discover/public/saved_searches/saved_search_alias_match_redirect.ts

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { useEffect } from 'react';
+import { i18n } from '@kbn/i18n';
+import { getSavedSearchFullPathUrl } from './saved_searches_utils';
+import { SAVED_SEARCH_TYPE } from './constants';
+
+import type { SavedSearch } from './types';
+import type { SpacesApi } from '../../../../../x-pack/plugins/spaces/public';
+
+interface SavedSearchAliasMatchRedirectProps {
+  savedSearch?: SavedSearch;
+  spaces?: SpacesApi;
+}
+
+export const useSavedSearchAliasMatchRedirect = ({
+  savedSearch,
+  spaces,
+}: SavedSearchAliasMatchRedirectProps) => {
+  useEffect(() => {
+    async function aliasMatchRedirect() {
+      if (savedSearch) {
+        const { aliasTargetId, outcome } = savedSearch.sharingSavedObject ?? {};
+
+        if (spaces && aliasTargetId && outcome === 'aliasMatch') {
+          await spaces.ui.redirectLegacyUrl(
+            getSavedSearchFullPathUrl(aliasTargetId),
```

Comment:
Seems like we do not need to include `/app/discover` here according to this [**line**](https://github.com/elastic/kibana/pull/107256/files#diff-2435b1c7a860e80fd5e604857bc798b3b0464a72a334005f35ea3cdecbaa6fe7R70) in the POC guide.

Reply:
thanks, checked that and short url also work